### ISSUE TITLE
Managed tuple packing and parsing

### ISSUE THREAD

**Issue by danthegoodman1** [danthegoodman1/ObjectKV#24]
See https://github.com/apple/foundationdb/blob/main/design/tuple.md as design inspiration.

By packing tuples with null termination, we can allow listing of prefixes without recursion (e.g. prefix scan `/dir` without seeing `/dir/a/1`)

**Comment by danthegoodman1** [comment]
```py
class PathEncoder:
    DIR_SEPARATOR = "/"
    NULL_BYTE = "\x00"
    BOUNDARY_BYTE = "\xFF"

    @staticmethod
    def encode_path(path: str) -> str:
        """
        Encode a path into a key that enables non-recursive directory listing.
        The format should be: /dir\x00/a\x00/1\x00 
        NOT: /\x00dir/\x00a\x00 (which was wrong)
        """
        if not path.startswith("/"):
            path = "/" + path

        # Remove trailing slash if present
        if path != "/" and path.endswith("/"):
            path = path[:-1]

        parts = path.split(PathEncoder.DIR_SEPARATOR)
        # Skip empty first part from leading slash
        parts = parts[1:]
        # Add null terminator after each component
        return PathEncoder.DIR_SEPARATOR + PathEncoder.NULL_BYTE.join(parts) + PathEncoder.NULL_BYTE

    @staticmethod
    def get_directory_range(directory: str) -> tuple[str, str]:
        """
        Returns (start_key, end_key) for range query to list immediate children.
        Uses \xFF as the boundary byte since it sorts after all valid path characters.
        """
        if not directory.startswith("/"):
            directory = "/" + directory

        if directory != "/" and directory.endswith("/"):
            directory = directory[:-1]

        # Start key is the directory path encoded
        start_key = PathEncoder.encode_path(directory)

        # End key uses 0xFF which sorts after any possible path component
        end_key = start_key + PathEncoder.BOUNDARY_BYTE

        return start_key, end_key

# Example usage and verification
encoder = PathEncoder()

# Store these keys in the KV store
paths = ["/dir", "/dir/a", "/dir/b", "/dir/a/1", "/dir/a/2"]
encoded_paths = [encoder.encode_path(p) for p in paths]

print("Encoded paths:")
for path, encoded in zip(paths, encoded_paths):
    print(f"{path:12} -> {encoded!r}")

# To list /dir contents
directory = "/dir"
start_key, end_key = encoder.get_directory_range(directory)

print(f"\nTo list '{directory}' contents, query range:")
print(f"start_key: {start_key!r}")
print(f"end_key:   {end_key!r}")

# Verify the lexicographic ordering works correctly
print("\nKeys in lexicographic order:")
sorted_keys = sorted([
    encoder.encode_path("/dir"),        # The directory itself
    encoder.encode_path("/dir/a"),      # Should be included
    encoder.encode_path("/dir/b"),      # Should be included
    encoder.encode_path("/dir/a/1"),    # Should not be included
    start_key,                          # Range start
    end_key,                           # Range end
])
for key in sorted_keys:
    print(f"  {key!r}")
```

**Comment by danthegoodman1** [comment]
Closed by #25
